applications/plugins/SofaImplicitField/components/engine/FieldToSurfaceMesh.h

```diff
@@ -0,0 +1,131 @@
+*                                                                             *
+* This program is distributed in the hope that it will be useful, but WITHOUT *
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or       *
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License *
+* for more details.                                                           *
+*                                                                             *
+* You should have received a copy of the GNU Lesser General Public License    *
+* along with this program. If not, see <http://www.gnu.org/licenses/>.        *
+*******************************************************************************
+* Authors: The SOFA Team and external contributors (see Authors.txt)          *
+*                                                                             *
+* Contact information: contact@sofa-framework.org                             *
+******************************************************************************/
+#pragma once
+#include <SofaImplicitField/config.h>
+#include <SofaImplicitField/components/geometry/ScalarField.h>
+
+#include <sofa/core/topology/BaseMeshTopology.h>
+#include <future>
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+namespace sofaimplicitfield::component::engine
+{
+using namespace sofa;
+
+typedef sofa::core::topology::BaseMeshTopology::SeqTriangles SeqTriangles;
+typedef sofa::core::topology::BaseMeshTopology::Triangle Triangle;
+typedef sofa::type::vector<sofa::type::Vec3d> VecCoord;
+
+using sofa::component::geometry::ScalarField;
+using sofa::core::visual::VisualParams ;
+using sofa::core::objectmodel::BaseObject ;
+using sofa::type::Vec3d ;
+
+class FieldToSurfaceMesh : public BaseObject
+{
+public:
+    SOFA_CLASS(FieldToSurfaceMesh, BaseObject);
+
+    virtual void init() override ;
+    virtual void draw(const VisualParams*params) override ;
+
+    double getStep() const { return mStep.getValue(); }
+    void setStep(double val) { mStep.setValue(val); }
+
+    double getIsoValue() const { return mIsoValue.getValue(); }
+    void setIsoValue(double val) { mIsoValue.setValue(val); }
+
+    const Vec3d& getGridMin() const { return mGridMin.getValue(); }
+    void setGridMin(const Vec3d& val) { mGridMin.setValue(val); }
+    void setGridMin(double x, double y, double z) { mGridMin.setValue( Vec3d(x,y,z)); }
+
+    const Vec3d& getGridMax() const { return mGridMax.getValue(); }
+    void setGridMax(const Vec3d& val) { mGridMax.setValue(val); }
+    void setGridMax(double x, double y, double z) { mGridMax.setValue( Vec3d(x,y,z)); }
+
+protected:
+    SingleLink<FieldToSurfaceMesh, ScalarField,
+               BaseLink::FLAG_STOREPATH|BaseLink::FLAG_STRONGLINK> l_field ;
+
+    Data <double > mStep;
+    Data <double > mIsoValue;
+
+    Data< Vec3d > mGridMin;
+    Data< Vec3d > mGridMax;
```

Comment:
```suggestion
    Data <double > d_mStep;
    Data <double > d_mIsoValue;

    Data< Vec3d > d_mGridMin;
    Data< Vec3d > d_mGridMax;
```
it is a new component so we should stick to the coding guidelines
(same for members with `m_` etc....